### PR TITLE
Fixes AI interaction with modular computers

### PR DESCRIPTION
**Pull request by Bokkiewokkie**
## About The Pull Request

Makes get_overmap() check for the z level of the location of an atom instead of just the z level of the atom, because inside containers your z level is 0.

## Why It's Good For The Game

Fixes: #1734

## Changelog
:cl:
fix: Fixed modular computer interaction for AI units.
/:cl:

### CODE REVIEW

**Comment by covertcorvid** [comment]
if loc is ever null this'll runtime

File: nsv13/code/__HELPERS/overmap.dm [hunk 1]
```diff
@@ -1,12 +1,12 @@
 /atom/proc/get_overmap() //Helper proc to get the overmap ship representing a given area.
 	RETURN_TYPE(/obj/structure/overmap)
-	if(!z)
+	if(loc && istype(loc, /obj/structure/overmap))
+		return loc
+	if(!loc?.z)
```
Comment (TheFakeElon) [inline nsv13/code/__HELPERS/overmap.dm:5]: if loc is null you'll probably want it to runtime
```suggestion
	if(isovermap(loc))
		return loc
	if(!loc.z)
```

**Comment by TheFakeElon** [comment]
This'll work for the majority of cases however, it can still runtime if it's located in containers recursively (i.e modular computer inside a bag inside a closet). You could use get_turf() to get around this

File: nsv13/code/__HELPERS/overmap.dm (hunk 1 above)
Comment (Bokkiewokkie) [reply nsv13/code/__HELPERS/overmap.dm:5]: > if loc is null you'll probably _want_ it to runtime, as something else has gone catastrophically wrong

The problem is there's plenty of things in nullspace calling get_overmap() for some reason and I'm gonna have to fix all of them then...

**Comment by Bokkiewokkie** [comment]
> This'll work for the majority of cases however, it can still runtime if it's located in containers recursively (i.e modular computer inside a bag inside a closet). You could use get_turf() to get around this

Tbh should just refactor most of this proc to not be several layers of lasange and spaghetti stacked on top of one another

File: nsv13/code/__HELPERS/overmap.dm (hunk 1 above)
Comment (TheFakeElon) [reply nsv13/code/__HELPERS/overmap.dm:5]: Better to have the runtime log than not though right? Null loc is (almost) always unexpected behavior

**Comment by TheFakeElon** [comment]
> Tbh should just refactor most of this proc to not be several layers of lasange and spaghetti stacked on top of one another

#1689 :sob:

**Pull request merged**